core/src/main/java/tachyon/master/TachyonMaster.java

```diff
@@ -186,20 +185,13 @@ boolean isZookeeperMode() {
     return mZookeeperMode;
   }
 
-  private void login() throws IOException {
-    MasterConf mConf = MasterConf.get();
-    if (mConf.KEYTAB == null || mConf.PRINCIPAL == null) {
-      return;
-    }
+  private void connect() throws IOException {
     UnderFileSystem ufs = UnderFileSystem.get(CommonConf.get().UNDERFS_ADDRESS);
-    if (ufs instanceof UnderFileSystemHdfs) {
-      ((UnderFileSystemHdfs) ufs).login(mConf.KEYTAB_KEY, mConf.KEYTAB, mConf.PRINCIPAL_KEY,
-          mConf.PRINCIPAL, NetworkUtils.getFqdnHost(mMasterAddress));
-    }
+    ufs.connectFromMaster(NetworkUtils.getFqdnHost(mMasterAddress));
```

Comment:
I didn't want to preclude that other future implementations might need to take some actions to establish a connection.  The actual login logic has moved to these methods in the HDFS implementation.

The logic can't be confined to the HDFS implementation constructor (i.e. rendering these methods unnecessary) because logging into HDFS requires knowing the host we want to present ourselves as.  While we could try and auto-detect the host name of the system there is no guarantee the host name we'd find is the externally visible host name which is what we actually need.